plugins/outputs/parquet/parquet.go

```diff
@@ -0,0 +1,339 @@
+				case arrow.PrimitiveTypes.Uint32:
+					builder.Field(index).(*array.Uint32Builder).AppendNull()
+				case arrow.PrimitiveTypes.Uint64:
+					builder.Field(index).(*array.Uint64Builder).AppendNull()
+				case arrow.PrimitiveTypes.Float32:
+					builder.Field(index).(*array.Float32Builder).AppendNull()
+				case arrow.PrimitiveTypes.Float64:
+					builder.Field(index).(*array.Float64Builder).AppendNull()
+				case arrow.BinaryTypes.String:
+					builder.Field(index).(*array.StringBuilder).AppendNull()
+				case arrow.FixedWidthTypes.Boolean:
+					builder.Field(index).(*array.BooleanBuilder).AppendNull()
+				default:
+					return nil, fmt.Errorf("unsupported type: %T", value)
+				}
+
+				continue
+			}
+
+			switch col.Type {
+			case arrow.PrimitiveTypes.Int8:
+				builder.Field(index).(*array.Int8Builder).Append(value.(int8))
+			case arrow.PrimitiveTypes.Int16:
+				builder.Field(index).(*array.Int16Builder).Append(value.(int16))
+			case arrow.PrimitiveTypes.Int32:
+				builder.Field(index).(*array.Int32Builder).Append(value.(int32))
+			case arrow.PrimitiveTypes.Int64:
+				builder.Field(index).(*array.Int64Builder).Append(value.(int64))
+			case arrow.PrimitiveTypes.Uint8:
+				builder.Field(index).(*array.Uint8Builder).Append(value.(uint8))
+			case arrow.PrimitiveTypes.Uint16:
+				builder.Field(index).(*array.Uint16Builder).Append(value.(uint16))
+			case arrow.PrimitiveTypes.Uint32:
+				builder.Field(index).(*array.Uint32Builder).Append(value.(uint32))
+			case arrow.PrimitiveTypes.Uint64:
+				builder.Field(index).(*array.Uint64Builder).Append(value.(uint64))
+			case arrow.PrimitiveTypes.Float32:
+				builder.Field(index).(*array.Float32Builder).Append(value.(float32))
+			case arrow.PrimitiveTypes.Float64:
+				builder.Field(index).(*array.Float64Builder).Append(value.(float64))
+			case arrow.BinaryTypes.String:
+				builder.Field(index).(*array.StringBuilder).Append(value.(string))
+			case arrow.FixedWidthTypes.Boolean:
+				builder.Field(index).(*array.BooleanBuilder).Append(value.(bool))
+			default:
+				return nil, fmt.Errorf("unsupported type: %T", value)
+			}
+		}
+	}
+
+	record := builder.NewRecord()
+	return record, nil
+}
+
+func (p *Parquet) createSchema(metrics []telegraf.Metric) (*arrow.Schema, error) {
+	rawFields := make(map[string]arrow.DataType, 0)
+	for _, metric := range metrics {
+		for _, field := range metric.FieldList() {
+			if _, ok := rawFields[field.Key]; !ok {
+				arrowType, err := goToArrowType(field.Value)
+				if err != nil {
+					return nil, fmt.Errorf("error converting '%s=%s' field to arrow type: %w", field.Key, field.Value, err)
+				}
+				rawFields[field.Key] = arrowType
+			}
+		}
+		for _, tag := range metric.TagList() {
+			if _, ok := rawFields[tag.Key]; !ok {
+				arrowType, err := goToArrowType(tag.Value)
+				if err != nil {
+					return nil, fmt.Errorf("error converting '%s=%s' tag to arrow type: %w", tag.Key, tag.Value, err)
+				}
+				rawFields[tag.Key] = arrowType
```

Comment:
You could directly use a string type here, can't you?

Reply:
ah yes! thank you